### ISSUE TITLE
5.12.13 - Cannot create property  '$isUnderneathDocArray' on string 'nested'

### ISSUE THREAD

**Issue by athompkins98** [Automattic/mongoose#10357]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Console output: "Cannot create property '$isUnderneathDocArray' on string 'nested'"

**If the current behavior is a bug, please provide the steps to reproduce.**
Run script below using mongoose 5.12.7+

```
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

const OutputSchema = new Schema({
      1: { left: { type: String }},
});

const AslSchema = new Schema({
  output: { type: OutputSchema }
});

const SessionSchema = new Schema({
  asls: { type: [AslSchema] }
});

const BlockSchema = new Schema({
  sessions: { type: [SessionSchema]  }
});
```

N/A

**What is the expected behavior?**
No output.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node: 15.14.0
Mongoose: 5.12.13
MongoDB: N/A for this instance.

Question: Why does the script work with older versions of Mongoose but not the latest release? I tried looking through the changelogs to see if I could find a solution, but was unable to.

**Comment by IslandRhythms** [comment]
```
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const OutputSchema = new Schema({
      1: { left: { type: String }},
});

const AslSchema = new Schema({
  output: { type: OutputSchema }
});

const SessionSchema = new Schema({
  asls: { type: [AslSchema] }
});

const BlockSchema = new Schema({
  sessions: { type: [SessionSchema]  }
});

const Test = mongoose.model('Test', BlockSchema);

async function test() {
    await mongoose.connect('mongodb://localhost:27017/test', {
        useNewUrlParser: true,
        useUnifiedTopology: true
    })
    await mongoose.connection.dropDatabase();

    await Test.create({sessions: {asls: [{output: {1: {left: 'Test'}}}]}})
}

test();
```

**Comment by vmo-khanus** [comment]
I slightly changed your example and proposed a fix here:
https://github.com/Automattic/mongoose/pull/10361

**Comment by vkarpov15** [comment]
Fixed by #10361

**Comment by bessfernandez** [comment]
Thanks for fixing! I had also run into this earlier in the week.
